### DISCUSSION TITLE
How to synchronize browser and backend clock?

### DISCUSSION THREAD

**Discussion by Evisolpxe** [reflex-dev/reflex#95] [Ideas]
I noticed clock example clock used `await asyncio.sleep(1)` to show timer.
But that is not exactly one second, there is an difference of about 10ms after network synchronization. 
The final performance is that it skips 1 second after running a while.

Of course we could use `sleep(0.1)` for query faster, but that's definite not best practice.
Main issue is we need have ability to operate BOM, like `setTimeout()`

**Comment by Alek99** [comment]
Yeah thanks for bringing this up. This is something I will think about, I agree this clock example with bad internet connection can be a little weird. We'll brainstorm some good way to potentially add this functionality.

**Reply by FHU-yezi** [reply]
Add a timestamp in server to client request?

**Reply by picklelo** [reply]
Yeah, we were thinking this... though it may be an issue since the server/client clocks may not be synchronized?

**Reply by FHU-yezi** [reply]
> Yeah, we were thinking this... though it may be an issue since the server/client clocks may not be synchronized?

Yes but we need to assume the server-side clock is accurate, or hacker can modify client clock and break the sync, jump something based on time etc.

If the clock is modified a lot, TLS handshake may be failed first.
